pkg/pillar/cmd/volumemgr/initialvolumestatus.go

```diff
@@ -28,61 +26,31 @@ import (
 var nilUUID = uuid.UUID{}
 
 // appRwVolumeName - Returns name of the image ( including parent dir )
-// Note that we still use the sha in the filename to not impact running images. Otherwise
-// we could switch this to imageID
-// XXX other types of volumes might want a different name.
-func appRwVolumeName(sha256, uuidStr string, purgeCounter uint32, format zconfig.Format,
-	origin types.OriginType, isContainer bool) string {
+func appRwVolumeName(uuidStr string, generationCounter uint32, isContainer bool) string {
 
-	purgeString := ""
-	if purgeCounter != 0 {
-		purgeString = fmt.Sprintf("#%d", purgeCounter)
-	}
+	purgeString := fmt.Sprintf("#%d", generationCounter)
 	if isContainer {
-		return fmt.Sprintf("%s-%s%s", sha256, uuidStr, purgeString)
+		return fmt.Sprintf("%s%s", uuidStr, purgeString)
 	}
-	if origin != types.OriginTypeDownload {
-		log.Fatalf("XXX unsupported origin %v", origin)
-	}
-	formatStr := strings.ToLower(format.String())
-	return fmt.Sprintf("%s/%s-%s%s.%s", rwImgDirname, sha256,
-		uuidStr, purgeString, formatStr)
+	return fmt.Sprintf("%s/%s%s", rwImgDirname, uuidStr, purgeString)
```

Comment:
Why did you have to add rwImgDirname here?

Reply:
Removed.